### PR TITLE
Add test ORC multi column data generator.

### PR DESCRIPTION
**Pull request by mx123**
## Description
Add test ORC multi column data generator.

Fixes: #15420

### CODE REVIEW

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 1]
```diff
@@ -971,16 +971,26 @@ public static DataSize writeOrcFileColumnHive(File outputFile, RecordWriter reco
             throws Exception
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:971]: `Write multi collumn data ORC file`

`collumn` -> `column`

Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:971]: tx. done

File: lib/trino-orc/src/test/java/io/trino/orc/TestOrcLz4.java [hunk 2]
```diff
@@ -15,29 +15,122 @@
...
+    private TempFile tempFile;
+
+    @BeforeClass
+    public void setUp()
+            throws Exception
+    {
+        tempFile = new TempFile();
+        Random random = new Random();
+        List<Iterator<?>> iterator = ImmutableList.of(
+                Stream.generate(() -> random.nextLong()).limit(POSITION_COUNT).iterator(),
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/TestOrcLz4.java:72]: use lambda for the random generators `random::nextLong`

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 3]
```diff
@@ -971,16 +971,26 @@ public static DataSize writeOrcFileColumnHive(File outputFile, RecordWriter reco
...
     {
         SettableStructObjectInspector objectInspector = createSettableStructObjectInspector("test", type);
+        return writeOrcFileColumnHive(outputFile, recordWriter, objectInspector,  ImmutableList.of(type), ImmutableList.of(values));
+    }
+
+    public static DataSize writeOrcFileColumnHive(File outputFile, RecordWriter recordWriter, SettableStructObjectInspector objectInspector, List<Type> types, List<Iterator<?>> data)
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:977]: This public method signature is too brittle at the moment.

`List<Iterator<?>> data` can be composed out of iterators of different length.


One option: What about grouping: `type` and his associated `data`/ `values` under a java `record`?

There is very likely a more elegant / less brittle way to expose the functionality of this method.

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 4]
```diff
@@ -989,6 +994,50 @@ public static DataSize writeOrcFileColumnHive(File outputFile, RecordWriter reco
         return succinctBytes(outputFile.length());
     }
 
+    public static void writeOrcHive(
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1018]: writeOrcHive -> writeOrcColumnsHive

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 5]
```diff
@@ -989,6 +995,70 @@ public static DataSize writeOrcFileColumnHive(File outputFile, RecordWriter reco
...
+            throws Exception
+    {
+        Object row = objectInspector.create();
+
+        List<StructField> fields = ImmutableList.copyOf(objectInspector.getAllStructFieldRefs());
+        Serializer serializer = new OrcSerde();
+
+        int fieldCount = fields.size();
+        while (values.hasNext()) {
+            Function<Integer, Object> itemFactory = values.next();
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1049]: nit `itemFactory` -> `valueGetter`

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 6]
```diff
@@ -989,6 +995,70 @@ public static DataSize writeOrcFileColumnHive(File outputFile, RecordWriter reco
...
+            for (int fieldIndex = 0; fieldIndex < fieldCount; fieldIndex++) {
+                Object value = itemFactory.apply(fieldIndex);
+                value = preprocessWriteValueHive(types.get(fieldIndex), value);
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1052]: we should validate that `types` has the same amount of elements as `fieldCount`

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 7]
```diff
@@ -964,10 +970,10 @@ public static DataSize writeOrcColumnHive(File outputFile, Format format, Compre
...
         RecordWriter recordWriter = createOrcRecordWriter(outputFile, format, compression, type);
-        return writeOrcFileColumnHive(outputFile, recordWriter, type, values);
+        return writeOrcColumnHive(outputFile, recordWriter, type, values);
     }
 
-    public static DataSize writeOrcFileColumnHive(File outputFile, RecordWriter recordWriter, Type type, Iterator<?> values)
+    public static DataSize writeOrcColumnHive(File outputFile, RecordWriter recordWriter, Type type, Iterator<?> values)
             throws Exception
     {
         SettableStructObjectInspector objectInspector = createSettableStructObjectInspector("test", type);
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:979]: why don't we reuse the new method you just added?

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 8]
```diff
@@ -964,10 +970,10 @@ public static DataSize writeOrcColumnHive(File outputFile, Format format, Compre
             throws Exception
     {
         RecordWriter recordWriter = createOrcRecordWriter(outputFile, format, compression, type);
-        return writeOrcFileColumnHive(outputFile, recordWriter, type, values);
+        return writeOrcColumnHive(outputFile, recordWriter, type, values);
     }
 
-    public static DataSize writeOrcFileColumnHive(File outputFile, RecordWriter recordWriter, Type type, Iterator<?> values)
+    public static DataSize writeOrcColumnHive(File outputFile, RecordWriter recordWriter, Type type, Iterator<?> values)
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:976]: pre-existing: `outputFile` does not need to be passed to this method.

`succinctBytes(outputFile.length())` can be calculated after the `writeOrcColumnHive` call.

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 9]
```diff
@@ -1155,6 +1225,32 @@ static RecordWriter createOrcRecordWriter(File outputFile, Format format, Compre
...
+    {
+        JobConf jobConf = new JobConf(newEmptyConfiguration());
+        OrcConf.WRITE_FORMAT.setString(jobConf, format == ORC_12 ? "0.12" : "0.11");
+        OrcConf.COMPRESS.setString(jobConf, compression.name());
+
+        Properties tableProperties = new Properties();
+        tableProperties.setProperty(OrcConf.STRIPE_SIZE.getAttribute(), "120000");
+        List<StructField> fields = ImmutableList.copyOf(objectInspector.getAllStructFieldRefs());
+        tableProperties.setProperty(IOConstants.COLUMNS, fields.stream()
+                .map(field -> field.getFieldName())
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1239]: nit: Use lambda method reference

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 4 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: done

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 5 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: done

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 9 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: done

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 6 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1060]: done

**Comment by ebyhr** [comment]
Could you rebase on upstream to resolve conflicts?

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 7 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:980]: due to it's actually similar with `writeOrcHiveSingleStringColumn` now but due to differs in `createOrcRecordWriter(File outputFile, Format format, CompressionKind compression, Type type)` with using `orc.bloom.filter` does not the compatible. for that it's need separate additional refactoring. to do that?

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 8 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:977]: `writeOrcColumnHive` methods atm are not changing due to https://github.com/trinodb/trino/pull/15434#discussion_r1053033720 so make sense to keep them as before.

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 3 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: outdated

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 8 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: done

**Comment by mx123** [comment]
> Could you rebase on upstream to resolve conflicts?

done

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 7 above)
Comment (findinpath) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:979]:
```
        SettableStructObjectInspector objectInspector = createSettableStructObjectInspector("test", type);

        writeOrcColumnsHive(
                recordWriter,
                objectInspector,
                ImmutableList.of(type),
                StreamSupport.stream(
                                Spliterators.spliteratorUnknownSize(values, Spliterator.ORDERED),
                                false)
                        .map(value -> (Function<Integer, Object>) index -> value).iterator());
```

I used in `io.trino.orc.OrcTester#writeOrcColumnsHive(org.apache.hadoop.hive.ql.exec.FileSinkOperator.RecordWriter, org.apache.hadoop.hive.serde2.objectinspector.SettableStructObjectInspector, java.util.List<io.trino.spi.type.Type>, java.util.Iterator<java.util.function.Function<java.lang.Integer,java.lang.Object>>)` 
 `SettableStructObjectInspector` instead of `StandardStructObjectInspector`

Comment (findinpath) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:979]: If this makes sense, please do remove as well `io.trino.orc.OrcTester#writeOrcHiveSingleStringColumn` method because it is not necessary anymore.

Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:979]: make sense. in progress.

Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:982]: done

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 10]
```diff
@@ -995,6 +996,26 @@ public static DataSize writeOrcColumnHive(File outputFile, RecordWriter recordWr
         return succinctBytes(outputFile.length());
     }
 
+    public static void writeOrcHiveSingleStringColumn(
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: We can probably inline this method on `TestCachingOrcDataSource` and squash this commit in the previous commit.

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 11]
```diff
@@ -985,7 +986,7 @@ public static void writeOrcColumnHive(RecordWriter recordWriter, Type type, Iter
                 (StandardStructObjectInspector) objectInspector,
                 ImmutableList.of(type),
                 StreamSupport.stream(
-                        Spliterators.spliteratorUnknownSize(values, Spliterator.ORDERED), false)
+                                Spliterators.spliteratorUnknownSize(values, Spliterator.ORDERED), false)
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:988]: The extra spacing is probably related to the previous commit.

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 12]
```diff
@@ -1215,43 +1216,87 @@ private static Object preprocessWriteValueHive(Type type, Object value)
...
-                new Path(outputFile.toURI()),
-                Text.class,
-                compression != NONE,
-                createTableProperties("test", getJavaObjectInspector(type).getTypeName()),
-                () -> {});
+        StandardStructObjectInspector objectInspector = getStandardStructObjectInspector(ImmutableList.of("test"), ImmutableList.of(getJavaObjectInspector(type)));
+
+        return RecordWriterBuilder.builder(outputFile, format, compression)
+                .putColumns(objectInspector)
+                .withBloomFilter(objectInspector, 0.50)
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1205]: This seems to be an addition compared to the previous version of the method. 
I see in the next commit that you remove the method `createTableProperties` - the two commits are probably related.

File: lib/trino-orc/src/test/java/io/trino/orc/TestOrcLz4.java [hunk 13]
```diff
@@ -52,13 +48,9 @@ public void setUp()
...
         Random random = new Random();
-        StandardStructObjectInspector objectInspector = getStandardStructObjectInspector(ImmutableList.of("x", "y", "z"),
-                ImmutableList.of(javaLongObjectInspector, javaIntObjectInspector, javaLongObjectInspector));
-        writeOrcColumnsHive(
-                tempFile.getFile(),
-                ORC_12,
-                LZ4,
-                objectInspector,
+        writeOrcColumnsHiveFile(
+                tempFile.getFile(), ORC_12, LZ4,
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/TestOrcLz4.java:52]: We either use all the parameters on one line or we use separate lines for each of the parameters.

**Review by findinpath — COMMENTED** [review]
I find the end result very good.

Please consider restructuring the commits.

I think that we can achieve the logical succession of the changes in fewer commits.
The refactoring related to working `RecordWriterBuilder` can be done as a preparatory commit and (if i'm not mistaken) the changes related to multi-columns orc writer support can be done in one commit.

File: lib/trino-orc/src/test/java/io/trino/orc/TestOrcLz4.java (hunk 13 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/TestOrcLz4.java]: ok. fixed.

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 12 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1205]: `Remove unused OrcTester#createTableProperties` squashed to `Add builder for ORC RecordWriter`

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 11 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: yeah, something went wrong with formatting. fixed.

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 10 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: seems, this commit (`Use multi column ORC file generator for TestCachingOrcDataSource`) is obsolete due to it was logically replaced with `Add ORC agnostic column amount write file method`. fixed - i've dropped the commit.

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 14]
```diff
@@ -966,14 +965,14 @@ private static List<Object> decodeRecordReaderStruct(Type type, List<?> fields)
         return newFields;
     }
 
-    public static DataSize writeOrcColumnHive(File outputFile, Format format, CompressionKind compression, Type type, Iterator<?> values)
+    public static void writeOrcColumnHive(File outputFile, Format format, CompressionKind compression, Type type, Iterator<?> values)
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:972]: `Remove unused outputFile in OrcTester#writeOrcColumnHive` -> `Remove unused return value`

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 15]
```diff
@@ -976,21 +979,15 @@ public static void writeOrcColumnHive(RecordWriter recordWriter, Type type, Iter
...
-            objectInspector.setStructFieldData(row, fields.get(0), value);
-
-            Writable record = serializer.serialize(row, objectInspector);
-            recordWriter.write(record);
-        }
-
-        recordWriter.close(false);
+        writeOrcColumnsHive(
+                recordWriter,
+                (StandardStructObjectInspector) objectInspector,
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:985]: why not use `SettableStructObjectInspector` as a parameter for `writeOrcColumnsHive` ?

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 16]
```diff
@@ -1194,59 +1195,92 @@ private static Object preprocessWriteValueHive(Type type, Object value)
...
-                tableProperties,
-                () -> {});
+        return RecordWriterBuilder.builder(outputFile, format, compression)
+                .putColumns(objectInspector)
+                .withStripeSize(120000L)
+                .build();
     }
 
-    static SettableStructObjectInspector createSettableStructObjectInspector(String name, Type type)
+    static class RecordWriterBuilder
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: `static class RecordWriterBuilder` -> `private static class RecordWriterBuilder`

**Comment by findinpath** [comment]
LGTM % comments

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 16 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: outdated, since commit squashed

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 14 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:972]: ok

File: lib/trino-orc/src/test/java/io/trino/orc/TestOrcLz4.java [hunk 17]
```diff
@@ -37,7 +37,7 @@ public void testReadLz4()
     {
         // this file was written with Apache ORC
         // TODO: use Apache ORC library in OrcTester
-        byte[] data = toByteArray(getResource("apache-lz4.orc"));
+        // byte[] data = toByteArray(getResource("apache-lz4.orc"));
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/TestOrcLz4.java]: unrelated

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 15 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: done

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 18]
```diff
@@ -960,33 +969,80 @@ private static List<Object> decodeRecordReaderStruct(Type type, List<?> fields)
...
         Serializer serializer = new OrcSerde();
 
         while (values.hasNext()) {
-            Object value = values.next();
-            value = preprocessWriteValueHive(type, value);
-            objectInspector.setStructFieldData(row, fields.get(0), value);
-
+            Function<Integer, Object> valueGetter = values.next();
+            for (int fieldIndex = 0; fieldIndex < fieldCount; fieldIndex++) {
+                Object value = valueGetter.apply(fieldIndex);
```
Comment (findinpath) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1037]: nit/optional: 

```
Object value = preprocessWriteValueHive(types.get(fieldIndex), valueGetter.apply(fieldIndex);
```

Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: make sense. replaced.

**Comment by mx123** [comment]
> I find the end result very good.
> 
> Please consider restructuring the commits.
> 
> I think that we can achieve the logical succession of the changes in fewer commits. The refactoring related to working `RecordWriterBuilder` can be done as a preparatory commit and (if i'm not mistaken) the changes related to multi-columns orc writer support can be done in one commit.

done.

**Comment by findinpath** [comment]
@mx123  there are minor checkstyle issues in the submission:

```
[INFO] There are 2 errors reported by Checkstyle 10.3.3 with checkstyle/airbase-checks.xml ruleset.
[ERROR] src/test/java/io/trino/orc/OrcTester.java:[991] (regexp) RegexpMultiline: Whitespace character before closing parenthesis
[ERROR] src/test/java/io/trino/orc/OrcTester.java:[1011] (regexp) RegexpMultiline: Whitespace character before closing parenthesis
```

**Comment by mx123** [comment]
> @mx123 there are minor checkstyle issues in the submission:
> 
> ```
> [INFO] There are 2 errors reported by Checkstyle 10.3.3 with checkstyle/airbase-checks.xml ruleset.
> [ERROR] src/test/java/io/trino/orc/OrcTester.java:[991] (regexp) RegexpMultiline: Whitespace character before closing parenthesis
> [ERROR] src/test/java/io/trino/orc/OrcTester.java:[1011] (regexp) RegexpMultiline: Whitespace character before closing parenthesis
> ```

@findinpath tx. it's weird, but locall build doesn't fail on that. fixed.

**Comment by findinpath** [comment]
Build is still red.

```
mvn clean install -P errorprone-compiler -DskipTests -nsu -pl :trino-orc
```

**Comment by mx123** [comment]
> mvn clean install -P errorprone-compiler -DskipTests -nsu -pl :trino-orc

i've fix file reading to bytest and tried locally this one^^^ - build passed with no errors.

**Comment by findinpath** [comment]
@ebyhr  CPTAL?

**Review by findinpath — APPROVED** [review]

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 19]
```diff
@@ -960,33 +969,77 @@ private static List<Object> decodeRecordReaderStruct(Type type, List<?> fields)
...
+    }
+
+    public static void writeOrcColumnsHiveFile(
+            File outputFile,
+            OrcTester.Format format,
+            CompressionKind compression,
+            List<String> names,
+            List<Type> types,
+            Iterator<Function<Integer, Object>> values)
+            throws Exception
```
Comment (ebyhr) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1000]: No change is requested. Throwing `Exception` is fine in a test method, but this is a helper method, not a test method.

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 20]
```diff
@@ -960,33 +969,77 @@ private static List<Object> decodeRecordReaderStruct(Type type, List<?> fields)
...
+        requireNonNull(types, "types is null");
+        requireNonNull(objectInspector, "objectInspector is null");
+
         Object row = objectInspector.create();
 
         List<StructField> fields = ImmutableList.copyOf(objectInspector.getAllStructFieldRefs());
+        int fieldCount = fields.size();
+        if (fieldCount != types.size()) {
+            throw new IllegalArgumentException("Field and type count must be the same");
+        }
```
Comment (ebyhr) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1028]: Use `checkArgument()` method.

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 21]
```diff
@@ -1142,33 +1195,92 @@ private static Object preprocessWriteValueHive(Type type, Object value)
...
+            OrcConf.WRITE_FORMAT.setString(jobConf, format == ORC_12 ? "0.12" : "0.11");
+            OrcConf.COMPRESS.setString(jobConf, compression.name());
+        }
+
+        public static RecordWriterBuilder builder(File file, Format format, CompressionKind compression)
+        {
+            return new RecordWriterBuilder(file, format, compression);
+        }
+
+        public RecordWriterBuilder putColumns(StandardStructObjectInspector objectInspector)
```
Comment (ebyhr) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1239]: `putColumns` sounds like adding columns, but this logic overwrites existing values if the method is called more than once. How about renaming to `withColumns` or moving to `builder` arguments?

**Review by ebyhr — APPROVED** [review]

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 19 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1000]: Actually it can be included (and it's included in other tests which didn't changed within PR) to test method, so it can be considered as a part of test. There are multiply exceptions are  throwing from:
```
Writable record = serializer.serialize(row, objectInspector); -> throws SerDeException
...
recordWriter.write(record); -> throws IOException
...
recordWriter.close(false); -> throws IOException;
```
do you propose wrap those to runtime exception?

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 20 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: ok, changed.

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 21 above)
Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: make sense. changed to `withColumns`.

**Comment by ebyhr** [comment]
@mx123 Could you avoid rebasing when there's no conflict? It makes it hard to see the diff since the last review in GitHub UI.

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java [hunk 22]
```diff
@@ -960,33 +970,75 @@ private static List<Object> decodeRecordReaderStruct(Type type, List<?> fields)
...
+            throws Exception
+    {
+        requireNonNull(types, "types is null");
+        requireNonNull(objectInspector, "objectInspector is null");
+
         Object row = objectInspector.create();
 
         List<StructField> fields = ImmutableList.copyOf(objectInspector.getAllStructFieldRefs());
+        int fieldCount = fields.size();
+        checkArgument(fieldCount != types.size(), "Field and type count must be the same");
```
Comment (ebyhr) [inline lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1027]:
```suggestion
        checkArgument(fieldCount == types.size(), "Field and type count must be the same");
```

Comment (mx123) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java]: thank you for finding - my bad, missed it :( . fixed.

**Comment by mx123** [comment]
> Could you avoid rebasing when there's no conflict? It makes it hard to see the diff since the last review in GitHub UI.

ok. sorry. will do.

**Review by findinpath — APPROVED** [review]

File: lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java (hunk 19 above)
Comment (ebyhr) [reply lib/trino-orc/src/test/java/io/trino/orc/OrcTester.java:1001]: > are you proposing wrap those to runtime exception?

No. The suggestion was throwing those exceptions as-is.

**Comment by ebyhr** [comment]
> apache-lz4.orc

Do you know what version created this file? Removing this file and using ORC library may mean losing test coverage for old versions.

**Comment by mx123** [comment]
> Can we test both `apache-lz4.orc` and generated ORC file by the library? Removing the file loses test coverage for a specific version.

ok. done:   https://github.com/trinodb/trino/compare/ae28767e093ebb4b1d9fd13bc91bfd63e18ff92a..0173236252814774e5b22d1fec46f4c40b317dec
